charmcraft/check-libraries/action.yaml

```diff
@@ -0,0 +1,70 @@
+name: "Check if charm libraries are updated"
+description: >
+  Check if - in the current pull request - charm libraries are updated, 
+  and add the relevant label to the PR.
+author: "Canonical"
+inputs:
+  path:
+    description: >
+      The location of the Charmcraft project.
+
+      Defaults to the base of the repository.
+    default: '.'
+  github_token:
+    description: >
+      The GitHub token to use to update the pull requests labels.
+    required: true
+  okay_label:
+    description: >
+      PR label to indicate that charm libraries are updated to their
+      latest LIBPATCH version.
+    default: "Charm Libraries: OK"
+  outdated_label:
+    description: >
+      PR label to indicate thta charm libraries are outdated, and new
+      versions can be fetched.
+    default: "Charm Libraries: Out of sync"
+  charmcraft_channel:
+    description: Install the Charmcraft snap from a specific channel.
+    default: 'latest/stable'
+
+outputs:
+  charms:
+    description: The names of any charm files created.
+    value: ${{ steps.pack-charm.outputs.charms }}
+  charmcraft-revision:
+    description: The revision of Charmcraft used for the pack.
+    value: ${{ steps.setup.outputs.charmcraft-revision }}
+  lxd-revision:
+    description: The revision of lxd used for the pack.
+    value: ${{ steps.setup.outputs.lxd-revision }}
+
+runs:
+  using: 'composite'
+  steps:
+    - id: setup
+      shell: bash
+      env:
+        CHANNEL: ${{ inputs.charmcraft_channel }}
+      run: |
+        sudo snap install charmcraft --classic --channel="$CHANNEL"
```

Comment:
One papercut here is that the charmcraft snap might already be installed, tracking a different channel, in which case the command succeeds but will silently be using the wrong thing
So at least [this part](https://github.com/canonical/craft-actions/blob/main/charmcraft/setup/action.yaml#L51) of the setup action I think we should 'borrow'